drivers/flash/flash_page_layout.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <flash.h>
+
+static int _flash_get_page_info(struct device *dev, off_t offs,
+				   bool use_addr, struct flash_pages_info *info)
+{
+	const struct flash_driver_api *api = dev->driver_api;
+	const struct flash_pages_layout *layout;
+	size_t page_count = 0;
+	off_t group_offs = 0;
+	u32_t num_in_group;
+	off_t end = 0;
+	size_t layout_size;
+
+	api->page_layout(dev, &layout, &layout_size);
```

Comment:
No - we don't expect that. This API returns pointer to "almost" constant array. This Array can be real constant or can be initialize once, at first call (depends on hardware). I can see sense in moving such a initialization to DEVICE_INIT section (for nrf5 I will move it to [nrf5_flash_init](https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/flash/soc_flash_nrf5.c#L200) function.